template/koa-rest-api/Dockerfile.dev-deps

```diff
@@ -1,11 +0,0 @@
-# syntax=docker/dockerfile:1.2
-
-FROM node:16-alpine AS dev-deps
-
-WORKDIR /workdir
-
-COPY package.json yarn.lock ./
-
-RUN \
-  --mount=type=secret,id=npm,dst=/workdir/.npmrc \
```

Comment:
I think we need the `syntax` and `mount` in the `deps` stage of the main Dockerfile now, or otherwise we can drop the `private-npm` Buildkite plugin. (By default the template does not use any private SEEK packages, but the pipeline is wired up to support that once you need it.)